src/Notes/WC_Admin_Note.php

```diff
@@ -504,6 +513,42 @@ public function clear_actions() {
 		$this->set_prop( 'actions', array() );
 	}
 
+	/**
+	 * Set note layout.
+	 *
+	 * @param string $layout Note layout.
+	 */
+	public function set_layout( $layout ) {
+		// If we don't receive a layout we will set it by default as "plain".
+		if ( empty( $layout ) ) {
+			$layout = 'plain';
+		}
+		$valid_layouts = array( 'banner', 'plain', 'thumbnail' );
+		if ( in_array( $layout, $valid_layouts, true ) ) {
+			$this->set_prop( 'layout', $layout );
+		} else {
+			$this->error( 'admin_note_invalid_data', __( 'The admin note date prop cannot be empty.', 'woocommerce-admin' ) );
```

Comment:
I like that you added some error handling here, but is the error message correct?

Reply:
Nice catch! I changed it in the commit [11c71cb](https://github.com/woocommerce/woocommerce-admin/pull/4262/commits/11c71cbb1003fcedc93b24911c97d1b25b1bbc0b).